src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -1161,6 +1399,169 @@ namespace
         return matchSuccess();
     }
 
+    REWRITER(NGSoftMaxOp)
```

Comment:
nit: can we remove those patterns into a separate .cpp file with a different name, or if not possible, just to the end of the this file with a clear comment to mention that they are call-back based ?